simple-search-engine-maven/src/main/java/searchengine/model/SearchDataset.java

```diff
@@ -0,0 +1,51 @@
+package searchengine.model;
+
+import lombok.Getter;
+
+import java.util.*;
+
+/**
+ * Class that stores data to search in and indexes for it
+ */
+@Getter
+public class SearchDataset {
+    private final Map<String, Set<Integer>> invertedIndexes;
+    private final List<String> lines;
+
+    public SearchDataset(List<String> lines) {
+        this.lines = lines;
+        this.invertedIndexes = this.invertIndexes();
+    }
+
+    public void update(String... lines) {
+        this.lines.addAll(Arrays.asList(lines));
+
+        Map<String, Set<Integer>> newIndexes = this.invertIndexes();
+        this.invertedIndexes.putAll(newIndexes);
+    }
+
+    public Map<String, Set<Integer>> getInvertedIndexes() {
+        return Collections.unmodifiableMap(invertedIndexes);
+    }
+
+    /**
+     * Indexing data in {@link SearchDataset}
+     *
+     * @return {@link Map} with all indexed data
+     */
+    private Map<String, Set<Integer>> invertIndexes() {
```

Comment:
update this method to be more functional (do not change the state of object but map the input) and naming have to be different (i.e. ``getInvertedIndex``)

by functional I mean this : 

```java
 private Map<String, Set<Integer>> invertIndexes(final List<String> input) {
....
}
```